src/python/pants/core/goals/test.py

```diff
@@ -524,6 +736,23 @@ async def _run_debug_tests(
     return Test(exit_code)
 
 
+@rule(desc="Determine environment for partition", level=LogLevel.DEBUG)
+async def get_environment(partition: TestRequest.SubPartition) -> EnvironmentName:
```

Comment:
Made this a `@rule` instead of a `@rule_helper` so I could `MultiGet` many of them at once.